### ISSUE TITLE
Describe timeline for packages to support of new minor releases of Python

### ISSUE THREAD

**Issue by namurphy** [heliophysicsPy/standards#20]
There are a few PyHC packages that can take a while to run on new releases of Python.  For example, it usually takes us over at PlasmaPy ∼4–6 months because we depend on Numba (which took ∼6 months to work on Python 3.11).  

What would we think of amending the standards to include a timeline for running on new versions of Python?  

My inclination would be to say that PyHC packages must work on (and be tested against) the newest version of Python within six months of its release. For the October 2023 release of Python 3.12, PyHC packages would then need to work on Python 3.12 by April 2024.  

Another possibility would be to make this a "must" for core packages and a "should" for all other packages, but I don't know if this would suffice if we want to have an up-to-date environment containing all PyHC packages, with all packages being tested. 🤔 Adding a test against the newest version of Python would also show that each PyHC package is getting the minimum level of maintenance on an annual basis.  We'd probably want to discuss this further before adopting it.

I'm also wondering if it would be worth recommending against core packages depending on Numba because of the delays associated with Numba becoming compatible with new Python releases.

**Comment by jibarnum** [comment]
@namurphy this smells deeply of a new PHEP, whenever you have the time...🙃

**Comment by sapols** [comment]
That PHEP already exists actually, PHEP-3 :)

**Comment by jibarnum** [comment]
@sapols lol @ me forgetting that that was included in PHEP-3. Then, as you mentioned on another issue, maybe this issue should be closed out by PHEP-3 once that PR is merged?
